framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts

```diff
@@ -0,0 +1,162 @@
+			expect(typeof result.data.data.header.version).toBe('number');
+			expect(typeof result.data.data.header.timestamp).toBe('number');
+			expect(typeof result.data.data.header.height).toBe('number');
+			expect(typeof result.data.data.header.previousBlockID).toBe('string');
+			expect(typeof result.data.data.header.transactionRoot).toBe('string');
+			expect(typeof result.data.data.header.generatorPublicKey).toBe('string');
+			expect(typeof result.data.data.header.reward).toBe('string');
+			expect(typeof result.data.data.header.asset.maxHeightPreviouslyForged).toBe('number');
+			expect(typeof result.data.data.header.asset.maxHeightPrevoted).toBe('number');
+			expect(typeof result.data.data.header.asset.seedReveal).toBe('string');
+		});
+
+		it('should respond with 404 and error message when block not found for specified height', async () => {
+			const { response, status } = await callNetwork(axios.get(getURL('/api/blocks/?height=2')));
+
+			expect(status).toBe(404);
+			expect(response).toEqual({
+				errors: [
+					{
+						message: "Block with height '2' was not found",
+					},
+				],
+			});
+		});
+
+		it('should respond with 400 and error message when height param is not number', async () => {
+			const { response, status } = await callNetwork(
+				axios.get(getURL('/api/blocks/?height=-nein-no')),
+			);
+
+			expect(status).toBe(400);
+			expect(response).toEqual({
+				errors: [{ message: 'The block height query parameter should be a number.' }],
+			});
+		});
+	});
+
+	describe('api/blocks/:blockID', () => {
+		it('should respond with block when block found for specified id', async () => {
+			const {
+				data: {
+					data: {
+						header: { id },
+					},
+				},
+			} = await axios.get(getURL('/api/blocks/?height=1'));
+			// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+			const result = await axios.get(getURL(`/api/blocks/${id}`));
+
+			expect(result.status).toBe(200);
+			expect(result).toHaveProperty('data');
+			expect(result.data).toHaveProperty('data.header');
+			expect(result.data).toHaveProperty('data.payload');
+			expect(result.data).toHaveProperty('data.header.id');
+			expect(result.data).toHaveProperty('data.header.version');
+			expect(result.data).toHaveProperty('data.header.timestamp');
+			expect(result.data).toHaveProperty('data.header.height');
+			expect(result.data).toHaveProperty('data.header.previousBlockID');
+			expect(result.data).toHaveProperty('data.header.transactionRoot');
+			expect(result.data).toHaveProperty('data.header.generatorPublicKey');
+			expect(result.data).toHaveProperty('data.header.reward');
+			expect(result.data).toHaveProperty('data.header.asset');
+			expect(result.data).toHaveProperty('data.header.asset.maxHeightPreviouslyForged');
+			expect(result.data).toHaveProperty('data.header.asset.maxHeightPrevoted');
+			expect(result.data).toHaveProperty('data.header.asset.seedReveal');
```

Comment:
if you do not like the above suggestion, then wrap the expectation to function and something like, blockValidation(result). and reuse the same above.

Reply:
Followed the other suggestion as right now are similar but maybe in the future it changes. Also I prefer not to have tests with much helpers that hide what's going on too much.